test/browser/component.js

```diff
@@ -843,3 +844,212 @@ describe(`Component with required attrs`, function () {
+    });
+
+    it(`returns same context as other components sharing the same root component`, async function () {
+      const darkApp = document.createElement(`slotted-dark-app`);
+      document.body.appendChild(darkApp);
+
+      const widget = document.createElement(`themed-widget`);
+      darkApp.appendChild(widget);
+
+      const siblingWidget = document.createElement(`slotted-light-app`);
+      darkApp.appendChild(siblingWidget);
+
+      const nephewWidget = document.createElement(`themed-widget`);
+      siblingWidget.appendChild(nephewWidget);
+
+      await nextAnimationFrame();
+      expect(widget.getContext(`theme`)).to.be.an.instanceof(DarkTheme);
+      expect(widget.getContext(`theme`)).to.equal(siblingWidget.getContext(`theme`));
+      expect(widget.getContext(`theme`)).to.equal(nephewWidget.getContext(`theme`));
+    });
+
+    it(`throws error when context name is not declared in config`, async function () {
+      const darkApp = document.createElement(`slotted-dark-app`);
+      document.body.appendChild(darkApp);
+
+      const widget = document.createElement(`default-light-themed-widget`);
+      darkApp.appendChild(widget);
+
+      await nextAnimationFrame();
+      expect(() => widget.getContext(`asdf`)).to.throw();
+    });
+
+    it(`throws error when context name is null or empty`, async function () {
+      const widget = document.createElement(`default-light-themed-widget`);
+      document.body.appendChild(widget);
+
+      await nextAnimationFrame();
+      expect(() => widget.getContext(``)).to.throw();
+      expect(() => widget.getContext()).to.throw();
+      expect(() => widget.getContext(null)).to.throw();
+    });
+
+    it(`throws error when called before component is connected`, async function () {
+      const widget = document.createElement(`default-light-themed-widget`);
+
+      await nextAnimationFrame();
+      expect(() => widget.getContext(`theme`)).to.throw();
+    });
+  });
+
+  describe(`lifecycle`, function () {
+    beforeEach(async function () {
+      document.body.innerHTML = ``;
+      await nextAnimationFrame();
+    });
+
+    it(`fails to connect when a context declared in config does not have a default context by itself or from any context ancestor`, async function () {
+      // modern browsers with native Custom Elements support will emit a global error event
+      const errors = [];
+      window.uncaughtErrorFilter = (errorEvent) => {
+        errors.push(errorEvent.message);
+        return true;
+      };
+
+      try {
+        const widget = document.createElement(`themed-widget`);
+        document.body.appendChild(widget);
+      } catch (err) {
+        // older browsers that rely on a Custom Elements polyfill will throw an error directly
+        errors.push(err.message);
+      }
+      await nextAnimationFrame();
+
+      expect(errors).to.have.lengthOf(1);
+      expect(errors[0]).to.contain(`context is not available`);
```

Comment:
can we check the more specific error message here? would this be it:
```suggestion
      expect(errors[0]).to.contain(`A "theme" context is not available`);
```